src/libmongoc/src/mongoc/mongoc-counters-private.h

```diff
@@ -48,8 +48,14 @@ _mongoc_counters_cleanup (void);
 static BSON_INLINE unsigned
 _mongoc_get_cpu_count (void)
 {
-#if defined(__linux__)
+#if defined(__linux__) && !defined(__ANDROID__)
    return get_nprocs ();
+#elif defined(__ANDROID__) && defined(_SC_NPROCESSORS_CONF)
+   long count = sysconf (_SC_NPROCESSORS_CONF);
+   if (count < 1) {
+      return 1;
+   }
+   return count;
```

Comment:
It looks like `sysconf()` is generally more portable on Linux systems, so it may be sufficient to replace the call to `get_nprocs()` entirely:

```suggestion
#if defined(__linux__) && defined(_SC_NPROCESSORS_CONF)
   long count = sysconf (_SC_NPROCESSORS_CONF);
   if (count < 1) {
      return 1;
   }
   return count;
```